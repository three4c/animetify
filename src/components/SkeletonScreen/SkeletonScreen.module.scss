.SkeletonScreen {
  & &__shimmer {
    position: relative;
    overflow: hidden;
    background-color: #eee;

    &::before {
      content: '';
      height: 100%;
      width: 100%;
      background: linear-gradient(
        90deg,
        transparent,
        rgba(#fff, 0.4),
        transparent
      );
      position: absolute;
      top: 0;
      left: 0;
      animation: skeleton 1.2s linear infinite;
    }
  }

  & &__title {
    width: 320px;
    height: 32px;
    margin-bottom: 48px;
  }

  & &__list {
    display: flex;
    flex-wrap: wrap;
  }

  & &__item {
    background-color: #fff;
    border-radius: 16px;
    box-shadow: 0 0 24px rgb(0 0 0 / 2%);
    padding: 24px;
    margin: 0 24px 32px 0;

    @media screen and (max-width: 1160px) {
      width: calc(100% / 3 - 24px * 2 / 3);

      &:nth-child(3n) {
        margin-right: 0;
      }
  
      &:nth-last-child(-n + 3) {
        margin-bottom: 0;
      }
    }

    @media (min-width: 1160px) and (max-width: 1456px) {
      width: calc(100% / 4 - 24px * 3 / 4);

      &:nth-child(4n) {
        margin-right: 0;
      }
  
      &:nth-last-child(-n + 4) {
        margin-bottom: 0;
      }
    }

    @media screen and (min-width: 1456px) {
      width: calc(100% / 5 - 24px * 4 / 5);

      &:nth-child(5n) {
        margin-right: 0;
      }
  
      &:nth-last-child(-n + 5) {
        margin-bottom: 0;
      }
    }

    &Time {
      width: 64px;
      height: 12px;
      margin-bottom: 8px;
    }

    &Title {
      height: 29px;
      margin-bottom: 32px;
    }

    &Tag {
      width: 48px;
      height: 20px;
      margin-bottom: 8px;
    }

    &Text {
      width: 50%;
      height: 19px;
      margin-bottom: 2px;
    }

    &SubText {
      height: 14px;

      &:not(:last-child) {
        margin-bottom: 16px;
      }
    }
  }
}

@keyframes skeleton {
  0% {
    transform: translateX(-100%);
  }
  100% {
    transform: translateX(100%);
  }
}
