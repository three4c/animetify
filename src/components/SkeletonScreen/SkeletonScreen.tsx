import React from 'react';

import styles from './SkeletonScreen.module.scss';

interface SkeletonScreenProps {
  /** 表示数 */
  length?: number;
}

const SkeletonScreen: React.FC<SkeletonScreenProps> = (props) => (
  <div className={styles.SkeletonScreen}>
    <div
      className={`${styles.SkeletonScreen__shimmer} ${styles.SkeletonScreen__title}`}
    />
    <ul className={styles.SkeletonScreen__list}>
      {[...Array(props.length)].map((_, index) => (
        <li className={styles.SkeletonScreen__item} key={index}>
          <div
            className={`${styles.SkeletonScreen__shimmer} ${styles.SkeletonScreen__itemTime}`}
          />
          <div
            className={`${styles.SkeletonScreen__shimmer} ${styles.SkeletonScreen__itemTitle}}`}
          />
          <div
            className={`${styles.SkeletonScreen__shimmer} ${styles.SkeletonScreen__itemTag}`}
          />
          <div
            className={`${styles.SkeletonScreen__shimmer} ${styles.SkeletonScreen__itemText}`}
          />
          <div
            className={`${styles.SkeletonScreen__shimmer} ${styles.SkeletonScreen__itemSubText}`}
          />
          <div
            className={`${styles.SkeletonScreen__shimmer} ${styles.SkeletonScreen__itemTag}`}
          />
          <div
            className={`${styles.SkeletonScreen__shimmer} ${styles.SkeletonScreen__itemText}`}
          />
          <div
            className={`${styles.SkeletonScreen__shimmer} ${styles.SkeletonScreen__itemSubText}`}
          />
        </li>
      ))}
    </ul>
  </div>
);

SkeletonScreen.defaultProps = {
  length: 9,
};

export default SkeletonScreen;
