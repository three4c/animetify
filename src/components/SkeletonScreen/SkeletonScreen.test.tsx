import React from 'react';

import { shallow } from 'enzyme';

import SkeletonScreen from './SkeletonScreen';

describe('<SkeletonScreen />', () => {
  it('SnapShot', () => {
    expect(shallow(<SkeletonScreen />)).toMatchSnapshot();
  });
});
