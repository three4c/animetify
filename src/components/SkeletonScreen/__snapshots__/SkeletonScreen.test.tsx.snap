// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkeletonScreen /> SnapShot 1`] = `
<div
  className="SkeletonScreen"
>
  <div
    className="SkeletonScreen__shimmer SkeletonScreen__title"
  />
  <ul
    className="SkeletonScreen__list"
  >
    <li
      className="SkeletonScreen__item"
      key="0"
    >
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTime"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTitle"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
    </li>
    <li
      className="SkeletonScreen__item"
      key="1"
    >
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTime"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTitle"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
    </li>
    <li
      className="SkeletonScreen__item"
      key="2"
    >
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTime"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTitle"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
    </li>
    <li
      className="SkeletonScreen__item"
      key="3"
    >
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTime"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTitle"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
    </li>
    <li
      className="SkeletonScreen__item"
      key="4"
    >
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTime"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTitle"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
    </li>
    <li
      className="SkeletonScreen__item"
      key="5"
    >
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTime"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTitle"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
    </li>
    <li
      className="SkeletonScreen__item"
      key="6"
    >
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTime"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTitle"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
    </li>
    <li
      className="SkeletonScreen__item"
      key="7"
    >
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTime"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTitle"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
    </li>
    <li
      className="SkeletonScreen__item"
      key="8"
    >
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTime"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTitle"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemTag"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemText"
      />
      <div
        className="SkeletonScreen__shimmer SkeletonScreen__itemSubText"
      />
    </li>
  </ul>
</div>
`;
